src/ape/api/providers.py

```diff
@@ -1041,30 +1040,26 @@ def prepare_transaction(self, txn: TransactionAPI) -> TransactionAPI:
         return txn
 
     def send_transaction(self, txn: TransactionAPI) -> ReceiptAPI:
-        if not txn.sender:
-            txn_hash = self.web3.eth.send_raw_transaction(txn.serialize_transaction())
 
-        else:
-            try:
-                account = self.account_manager[txn.sender]
-                if isinstance(account, ImpersonatedAccount):
-                    txn_dict = txn.dict()
-                    txn_params = cast(TxParams, txn_dict)
-                    txn_hash = self.web3.eth.send_transaction(txn_params)
-                else:
-                    txn_hash = self.web3.eth.send_raw_transaction(txn.serialize_transaction())
-            except ValueError as err:
-                vm_err = self.get_virtual_machine_error(err, txn=txn)
-
-                if "nonce too low" in str(vm_err):
-                    # Add additional nonce information
-                    new_err_msg = f"Nonce '{txn.nonce}' is too low"
-                    raise VirtualMachineError(
-                        new_err_msg, base_err=vm_err.base_err, code=vm_err.code
-                    )
-
-                vm_err.txn = txn
-                raise vm_err from err
+        try:
+            if txn.signature:
+                txn_hash = self.web3.eth.send_raw_transaction(txn.serialize_transaction())
+            else:
+                if (
+                    txn.sender not in self.chain_manager.provider.web3.eth.accounts  # type: ignore # noqa
+                ):
+                    self.chain_manager.provider.unlock_account(txn.sender)
+                txn_dict = txn.dict()
+                txn_params = cast(TxParams, txn_dict)
+                txn_hash = self.web3.eth.send_transaction(txn_params)
+        except ValueError as err:
+            vm_err = self.get_virtual_machine_error(err, txn=txn)
+            if "nonce too low" in str(vm_err):
```

Comment:
Should this part should occur within `def get_virtual_machine_error()`? That way, this custom error happens in all the necessary spots.

Reply:
added